swri_transform_util/src/swri_transform_util/wgs84_transformer.py

```diff
@@ -0,0 +1,108 @@
+EARTH_EQUATOR_RADIUS = 6378137.0
+
+
+class Wgs84Transformer(object):
+    """
+    A simple class for transforming from the WGS84 frame to a local_xy frame 
+    and vice versa.
+
+    Transforms to and from WGS84 and the local map frame using a local origin
+    as a reference.
+    """
+
+    def __init__(self, local_origin):
+        """
+        Constructor for the Wgs84Transformer class
+        :param geometry_msgs.Pose local_origin: An initialized local origin
+        """
+        self._reference_heading = -1.0 * euler_from_quaternion(quaternion=(local_origin.pose.orientation.x,
+                                                                           local_origin.pose.orientation.y,
+                                                                           local_origin.pose.orientation.z,
+                                                                           local_origin.pose.orientation.w))[2]  # get yaw from quaternion
+        self._cos_heading = math.cos(self._reference_heading)
+        self._sin_heading = math.sin(self._reference_heading)
+        self._reference_latitude = local_origin.pose.position.y * math.pi / 180
+        self._reference_longitude = local_origin.pose.position.x * math.pi / 180
+        self._depth = -local_origin.pose.position.z
+
+        p = EARTH_ECCENTRICITY * math.sin(self._reference_latitude)
+        p = 1.0 - p**2
+
+        rho_e_num = EARTH_EQUATOR_RADIUS * (1.0 - EARTH_ECCENTRICITY**2)
+        rho_e_den = math.sqrt(p) * p
+        rho_e = rho_e_num / rho_e_den
+        rho_n = EARTH_EQUATOR_RADIUS / math.sqrt(p)
+
+        self._rho_lat = rho_e - self._depth
+        self._rho_lon = (rho_n - self._depth) * math.cos(self._reference_latitude)
+
+    def wgs84_to_local_xy(self, wgs84_points):
+        """
+        Transforms point(s) in the WGS84 coordinate frame to the local_xy frame.
+        :param list wgs84_points: pair-wise list of (latitude, longitude) coordinates
+        :return: A pair-wise list of (x, y) coordinates in the local_xy frame
+        """
+        wgs84_points = np.array(wgs84_points)
+
+        r = wgs84_points * math.pi / 180.0
+
+        d = (r - [self._reference_latitude, self._reference_longitude]) * [self._rho_lat, self._rho_lon]
+
+        points = d.dot([[-self._sin_heading, self._cos_heading], [self._cos_heading, self._sin_heading]])
+
+        return points
+
+    def local_xy_to_wgs84(self, local_points):
+        """
+        Transforms point(s) in the local_xy frame to the WGS84 coordinate frame
+        :param list local_points: pair-wise list of (x, y) coordinates
+        :return: A pair-wise list of (latitude, longitude) coordinates in the WGS84 frame
+        """
+        points = np.array(local_points)
+
+        d = points.dot([[self._sin_heading * self._cos_heading, self._cos_heading],
+                        [self._sin_heading**2, self._sin_heading]])
+        d[:,0] = (points[:,1] - d[:,0]) / self._cos_heading
```

Comment:
PEP 8 also requires spaces after commas.
```suggestion
        d[:, 0] = (points[:, 1] - d[:, 0]) / self._cos_heading
```